hnn_core/network.py

```diff
@@ -432,9 +438,13 @@ def __eq__(self, other):
                                           target_gids=target_gids,
                                           loc=conn['loc'],
                                           receptor=conn['receptor'])
-            if len(match_conns) == 0:
+            if len(match_conns) == 1:
+                # check if connections are equal
+                if conn != other.connectivity[match_conns[0]]:
+                    return False
+            else:
```

Comment:
@ntolley I've made a change here to make sure it is comparing the connectivity contents between the to networks. Is there a possibility where the length of `match_conns` is greater than 1? Right now it will just return a False for those instances. Is that OK?

Reply:
It would be uncommon but it is totally possible to have two `_Connectivity` objects that are the same on all attributes searched by `pick_connection`.

It's not something that would frequently come up, but there's good reasons not to prevent people from calling `add_connection` multiple times

Reply:
Ah ok. So the current logic first checks that the new networks objects have equal number of connections. Once that clears it goes on to check the equivalency of connections using the pick_connection to find matches. Does the pick_connection match on all connection attributes or just a subset? In other words could c1 match with c2 but not actually be equivalent for all attributes?